### DISCUSSION TITLE
Preserve All Exif/XMP metadata

### DISCUSSION THREAD

**Discussion by ghost** [saucecontrol/PhotoSauce#80] [Q&A] [answered]
Do you have an example on how to preserve the GPS meta data? as after running the process it strips them all out. There are no clear examples (From what we can see) on what metadatanames we need to add to the collection as part of the proccessImageSettings to carry over them to the output stream.

**Comment by saucecontrol** [comment, marked as answer]
GPS is a tricky one.  The docs for the [Windows photo metadata policies](https://docs.microsoft.com/en-us/windows/win32/wic/system-gps) list values like `System.GPS.Latitude`, but these cannot be used directly.  There's a note in the docs for that one that explains:

> This value can be written by writing to `System.GPS.LatitudeNumerator` and `System.GPS.LatitudeDenominator`. It cannot be written directly. Values from different schemas are reconciled.

If you split them out as described, it should work.  For example, the following works for preserving GPS lat/long from an iPhone photo.

```C#
MetadataNames = new[] {
  "System.GPS.LatitudeNumerator",
  "System.GPS.LatitudeDenominator",
  "System.GPS.LatitudeRef",
  "System.GPS.LongitudeNumerator",
  "System.GPS.LongitudeDenominator",
  "System.GPS.LongitudeRef"
}
```
